### DISCUSSION TITLE
WARING in GRIB2_ipdtmpl_index setting

### DISCUSSION THREAD

**Discussion by BinbinZhou-NOAA** [dtcenter/METplus#2417] [Configuration] [answered]
We got WARNING in one of our EVS precip verification when reading CCPA climo data: 
 WARNING: Dictionary::lookup_num_array() -> numeric array lookup failed for name "GRIB2_ipdtmpl_index"WARNING:
WARNING:
WARNING: Dictionary::lookup_num_array() -> numeric array lookup failed for name "GRIB2_ipdtmpl_val"
WARNING:
WARNING:
WARNING: MetGrib2DataFile::data_plane_array() -> Using the first of 10 exact matching records for "APCP/L0": WARNING: record 1 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, -1
WARNING: record 2 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 0
WARNING: record 3 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 1000 
WARNING: record 4 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 2000
WARNING: record 5 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 5000
WARNING: record 6 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 10000 
WARNING: record 7 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 20000
WARNING: record 8 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 25000
WARNING: record 9 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 35000
WARNING: record 10 field 1: ipdtmpl[22] = 1, 8, 2, 0, 184, 0, 0, 1, 0, 1, 0, 0, 255, 0, 0, 0, 0, 1, 0, 0, 3, 50000
WARNING: Try setting the 0-based "GRIB2_ipdtmpl_index" and "GRIB2_ipdtmpl_val" config file options to select one.
WARNING:

In the conf file, have following settings:
GRID_STAT_CLIMO_MEAN_FIELD =  { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="-1"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="0"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="1000"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="2000"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="5000"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="10000"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="20000"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="25000"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="35000"; }, { name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="50000"; } 
We think the WARNINGs. came from the above setting.
But counting the index number from 0 one by one  (example):
WARNING: record 3 field 1: ipdtmpl[22] = 1,  8, 2, 0, 184,  0, 0,  1, 0, 1,  0,  0, 255, 0,  0,   0,   0, 1,  0,   0, 3, **1000** 
                                                                    0  1  2  3     4    5  6  7  8  9 10 11  12  13  14 15 16 17 18 19 20   21
the index of "1000" is 21. So the setting should be correct. Why the run got the WARNINGs.

Thanks!

Binbin Zhou 
EMC/NCEP

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @BinbinZhou-NOAA I see you're getting a warning related to the `GRIB2_ipdtmpl_index` and `GRIB2_ipdtmpl_val` configuration options. I tried some things and was able to replicate the warning message you're getting. The solution is just removing the enclosing quotes (") around the values.

The MET code expects them to be set as, and parses them as, integers. The warning message occurs because you're defining them as strings.

For example, just changes this setting FROM THIS:
```
{ name="APCP"; level="L0"; GRIB2_ipdtmpl_index="21"; GRIB2_ipdtmpl_val="0"; }
```
TO THIS:
```
{ name="APCP"; level="L0"; GRIB2_ipdtmpl_index=21; GRIB2_ipdtmpl_val=0; }
```
Please scan through your configuration settings and remove the quotes from around all of those `GRIB2_ipdtmpl_index` and `GRIB2_ipdtmpl_val` values.

For completeness sake, let me point you to [this section](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#fcst) of the MET User's Guide that describes these settings:

The “GRIB2_ipdtmpl_index” and “GRIB2_ipdtmpl_val” entries are arrays of integers which specify the product description template values to be used. The indices are 0-based. For example, use the following to request a GRIB2 record whose 9-th and 27-th product description template values are 1 and 2, respectively:
`GRIB2_ipdtmpl_index=[8, 26]; GRIB2_ipdtmpl_val=[1, 2];`

I'll note that MET actually expects them to be set as *arrays of integers* with square brackets around the list of values. However, the config file parsing code is smart enough to see a single value specified and store it in an array of length one. If you do ever need to specify multiple indices and corresponding values, you can do so using the square bracket array syntax.

**Comment by BinbinZhou-NOAA** [comment]
John,

  Got it.

Thanks.

Binbin
